src/test/groovy/CountryFactoryTest.groovy

```diff
@@ -0,0 +1,45 @@
+import com.covid.api.dto.CountryData
+import com.covid.api.dto.CovidSummaryResponse
+import com.covid.api.dto.GlobalData
+import com.covid.api.factory.CountryFactory
+import spock.lang.Specification
+
+import java.time.LocalDateTime
+
+class CountryFactoryTest extends Specification {
+
+    def factory = new CountryFactory()
+
+    def "assembleCountries() should return a list of countries with correct data"() {
+        when:
+        def summary = new CovidSummaryResponse()
+        summary.setId("id")
+        summary.setGlobal(new GlobalData(1, 2, 3, 4, 5, 6, "2023-01-26T00:00:00Z"))
+        summary.setCountries([new CountryData("id2", "US", "US", "united-states", 7, 8, 9, 10, 11, 12, "2023-01-26T00:00:00Z")])
+        summary.setDate("2023-01-26T00:00:00Z")
+        def countries = factory.assembleCountries(summary)
+
+        then:

```

Comment:
please check this link:
https://spockframework.org/spock/docs/1.0/data_driven_testing.html

1. try to use data table or data pipes
2. try to check the number of the calls on a mock object (e.g. WebRequest in the next test)